relay-metrics/src/aggregator.rs

```diff
@@ -487,6 +644,84 @@ impl Aggregator {
         self.cost_tracker.totals_cost_exceeded(max_total_cost)
     }
 
+    /// Split the provided buckets into batches and process each batch with the given function.
+    ///
+    /// For each batch, log a histogram metric.
+    pub fn process_batches(
+        &self,
+        buckets: impl IntoIterator<Item = Bucket>,
+        process: impl Fn(Vec<Bucket>),
```

Comment:
also not related, but i find it odd to have a generic `process` when we only call it one. It seems a bit like premature generalization to me